adam/experiment/observer.py

```diff
@@ -85,6 +88,160 @@ def report(self) -> None:
         )
 
 
+CSS = """
+body {
+    font-size: 1em;
+    font-family: sans-serif;
+}
+
+table td { p
+    padding: 1em; 
+    background-color: #FAE5D3 ;
+}
+"""
+
+
+@attrs(slots=True)
+class HTMLLogger:
+
+    output_dir: Path = attrib(validator=instance_of(Path), kw_only=True)
+    experiment_name: str = attrib(validator=instance_of(str), kw_only=True)
+    curriculum_name: str = attrib(validator=instance_of(str), kw_only=True)
+    logging_dir: Path = attrib(init=False)
+    outfile: io = attrib(init=False)
+    html_dumper: CurriculumToHtmlDumper = attrib(init=False)
+
+    def __attrs_post_init__(self):
+        self.logging_dir = self.output_dir / self.experiment_name
```

Comment:
Use `__attrs_post__init__` just for validation. If you need to do work at initialization-time, prefer object creation via a factory method.

Reply:
What do you mean by a factory method? __init__()?

Reply:
Something like: